File: recovered/src/com/lumiyaviewer/lumiya/ui/settings/RingtonePreference.java
package com.lumiyaviewer.lumiya.ui.settings;

import android.content.Context;
import android.content.res.TypedArray;
import android.media.Ringtone;
import android.media.RingtoneManager;
import android.net.Uri;
import android.support.v7.preference.Preference;
import android.util.AttributeSet;
import com.google.common.base.Objects;
import com.lumiyaviewer.lumiya.R;
import com.lumiyaviewer.lumiya.ui.media.NotificationSounds;

/* loaded from: classes.dex */
public class RingtonePreference extends Preference {
    private int defaultRawResource;

    public RingtonePreference(Context context) {
        super(context);
        this.defaultRawResource = 0;
    }

    public RingtonePreference(Context context, AttributeSet attributeSet) {
        super(context, attributeSet);
        this.defaultRawResource = 0;
        applyAttributes(context, attributeSet, 0, 0);
    }

    public RingtonePreference(Context context, AttributeSet attributeSet, int i) {
        super(context, attributeSet, i);
        this.defaultRawResource = 0;
        applyAttributes(context, attributeSet, i, 0);
    }

    public RingtonePreference(Context context, AttributeSet attributeSet, int i, int i2) {
        super(context, attributeSet, i, i2);
        this.defaultRawResource = 0;
        applyAttributes(context, attributeSet, i, i2);
    }

    private void applyAttributes(Context context, AttributeSet attributeSet, int i, int i2) {
        TypedArray obtainStyledAttributes = context.getTheme().obtainStyledAttributes(attributeSet, R.styleable.RingtonePreference, i, i2);
        try {
            this.defaultRawResource = obtainStyledAttributes.getResourceId(0, this.defaultRawResource);
        } finally {
            obtainStyledAttributes.recycle();
        }
    }

    int getDefaultRawResource() {
        return this.defaultRawResource;
    }

    @Override // android.support.v7.preference.Preference
    public CharSequence getSummary() {
        String string = getSharedPreferences().getString(getKey(), null);
        if (string == null) {
            return "Default";
        }
        Uri parse = Uri.parse(string);
        if (Objects.equal(NotificationSounds.getResourceUri(this.defaultRawResource), parse)) {
            return "Default";
        }
        if (string.isEmpty()) {
            return "Silent";
        }
        Ringtone ringtone = RingtoneManager.getRingtone(getContext(), parse);
        return ringtone != null ? ringtone.getTitle(getContext()) : "No sound selected";
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/ui/settings/RingtonePreference.kt
package com.lumiyaviewer.lumiya.ui.settings

import android.content.Context
import android.media.RingtoneManager
import android.net.Uri
import android.util.AttributeSet
import androidx.preference.Preference
import com.google.common.base.Objects
import com.lumiyaviewer.lumiya.R
import com.lumiyaviewer.lumiya.ui.media.NotificationSounds

class RingtonePreference : Preference {
    private var defaultRawResource = 0

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        applyAttributes(context, attrs, 0, 0)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        applyAttributes(context, attrs, defStyleAttr, 0)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) :
        super(context, attrs, defStyleAttr, defStyleRes) {
        applyAttributes(context, attrs, defStyleAttr, defStyleRes)
    }

    private fun applyAttributes(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) {
        val typedArray = context.theme.obtainStyledAttributes(attrs, R.styleable.RingtonePreference, defStyleAttr, defStyleRes)
        try {
            defaultRawResource = typedArray.getResourceId(0, defaultRawResource)
        } finally {
            typedArray.recycle()
        }
    }

    fun getDefaultRawResource(): Int = defaultRawResource

    override fun getSummary(): CharSequence? {
        val value = sharedPreferences?.getString(key, null) ?: return "Default"
        val parsed = Uri.parse(value)
        if (Objects.equal(NotificationSounds.getResourceUri(defaultRawResource), parsed)) {
            return "Default"
        }
        if (value.isEmpty()) {
            return "Silent"
        }
        val ringtone = RingtoneManager.getRingtone(context, parsed)
        return ringtone?.getTitle(context) ?: "No sound selected"
    }
}
